pkg/server/proto/server.proto

```diff
@@ -619,6 +623,12 @@ message Variable {
     HclPos start = 2;
     HclPos end = 3;
   }
+
+  message Ref {
+    string value = 2;
+    string type = 3;
+    string source = 4;
```

Comment:
String types spook me out. Maybe they're right, but here to just challenge the idea.

* Can `source` be an enum?
* I think you can encode the type in the type system, the same way it is for `Variable`. Example in the previous comment.